test/tools/router_check/validation.proto

```diff
@@ -0,0 +1,99 @@
+syntax = "proto3";
+
+package envoy.RouterCheckToolSchema;
+
+import "envoy/api/v2/core/base.proto";
+
+// [#protodoc-title: RouterCheckTool Validation]
+
+// The Validation Schema of the envoy router check test files.
+// The accepted input formats for the test are json and yaml.
+// The tool transparently converts json/yaml into this proto schema.
+message Validation {
+  // A collection of test cases.
+  repeated ValidationItem tests = 1;
+}
+
+// Schema for each test case.
+message ValidationItem {
+  // Name of the test case. There is no uniqueness constraint among the test case names.
+  string test_name = 1;
+
+  // The input constraints of the test case.
+  ValidationInput input = 2;
+
+  // The validations that need to be performed on the resultant route.
+  ValidationAssert validate = 3;
+}
+
+// Input values sent to the router that determine the returned route.
+// This includes the `pseudo-header <https://http2.github.io/http2-spec/#HttpRequest>`_ fields
+// defined in HTTP2.
+message ValidationInput {
+  // This pseudo-header field includes the authority portion of the target URI.
+  // Clients that generate HTTP/2 requests directly SHOULD use the :authority pseudo-header field
+  // instead of the Host header field.
+  string authority = 1 [json_name = ":authority"];
```

Comment:
I realize you are trying to replicate the existing schema, but IMO I would drop the `json_name` attributes in this file and just switch to names without `:` in them. I think it's less magical and since we are deprecating anyway might as well switch? @hennna any objection here?